pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -581,3 +599,137 @@ func installNetworkPolicies(instance *servingv1alpha1.KnativeServing) error {
 	}
 	return nil
 }
+
+func exposeMetricToServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	// getOperatorNamespace return namespace where knative-serving-operator has been installed
+	namespace, err := getOperatorNamespace()
+	if err != nil {
+		log.Info(fmt.Sprintf("empty %s because operator dont want to expose metric or running locally", knativeServingInstalledNamespace))
+		return nil
+	}
+
+	log.Info("exposing metric for installed knative version")
+	knativeVersion.WithLabelValues(version.Version, namespace).Set(float64(time.Now().Unix()))
+
+	// Get service knative-serving-operator in order to create service monitor
+	svcDetail, err := getOperatorService(namespace)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			log.Info(fmt.Sprintf("service knative-serving-operator does not exist in %s namespace", namespace))
+			return nil
+		}
+		return err
+	}
+
+	// Add label openshift.io/cluster-monitoring to namespace
+	if err = addMonitoringLabelToNamespace(namespace); err != nil {
+		return err
+	}
+
+	// CreateServiceMonitors creates ServiceMonitors objects based on Service objects.
+	if _, err := metric.CreateServiceMonitors(cfg, namespace, []*v1.Service{svcDetail}); err != nil {
+		if strings.Contains(err.Error(), "already exists") {
```

Comment:
This should also be an apimachinery error, please use https://godoc.org/k8s.io/apimachinery/pkg/api/errors#IsAlreadyExists